### PR TITLE
frontend: Add IP range start and end addresses to CIDR tooltip

### PR DESCRIPTION
**Pull request by kyoto**
screenshot-1

### CODE REVIEW

File: installer/frontend/components/cidr.jsx [hunk 1]
```diff
@@ -16,14 +17,35 @@ export const cidrSize = cidr => {
   return Math.pow(2, 32 - parseInt(bits, 10));
 };
 
+// Convert an IPv4 string to an integer
+const ip2int = ip => {
+  const [a, b, c, d] = ip.split('.').map(i => parseInt(i, 10));
+  return a * (256 ** 3) + b * (256 ** 2) + c * 256 + d;
```
Comment (squat) [inline installer/frontend/components/cidr.jsx]: another, faster way to do this, I think, would be to bitshift the pieces and `|` them together.

File: installer/frontend/components/cidr.jsx [hunk 2]
```diff
@@ -16,14 +17,35 @@ export const cidrSize = cidr => {
...
+};
+
+// Convert an integer to an IPv4 string
+const int2ip = int => [0, 1, 2, 3].map(i => Math.floor(int / (256 ** i)) % 256).reverse().join('.');
+
+// Get the first IP address in a CIDR range
+const cidrStart = cidr => {
+  const [ip, bits] = cidr.split('/');
```
Comment (squat) [inline installer/frontend/components/cidr.jsx]: should you `parseInt` here?

Comment (kyoto) [reply installer/frontend/components/cidr.jsx]: Good catch, but I discovered another bug in `cidrStart()` and ended up changing it anyway.

File: installer/frontend/components/cidr.jsx (hunk 1 above)
Comment (kyoto) [reply installer/frontend/components/cidr.jsx]: I checked this again, but unfortunately we do run into the signed 32bit int issue, so I it this as is.

**Comment by kyoto** [comment]
@squat I ended up reworking this by
- Rewriting `cidrStart()` to avoid another problem with bit shifting signed ints
- Moving all the CIDR helpers to new file `cidr.js`
- Adding unit tests

Could you please look again.

**Comment by kyoto** [comment]
cc @tlwu2013 @sym3tri

**Pull request merged**